Tasks/Common/utility-common/kubectlutility.ts

```diff
@@ -45,7 +46,22 @@ export async function downloadKubectl(version: string) : Promise<string> {
     }
     
     var kubectlPath = path.join(cachedToolpath, kubectlToolName + getExecutableExtention());
+    
+    if (!cachedToolpath || !fs.existsSync(kubectlPath))
+    {
+        var kubectlPathTmp = path.join(getTempDirectory(), kubectlToolName + getExecutableExtention());
+        tl.cp(KubectlDownloadPath, kubectlPathTmp, "-f");
+        fs.chmodSync(kubectlPathTmp, "777");
+        return kubectlPathTmp;
+    }
+
     fs.chmodSync(kubectlPath, "777");
+    
+    // prepend the tools path. instructs the agent to prepend for future tasks. Do this only if caching is successful
+    if(!process.env['PATH'].toLowerCase().startsWith(path.dirname(kubectlPath.toLowerCase()))) {
```

Comment:
we should do this even if caching is not successful.

Reply:
will do